assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * `core/modules` data store: module sharing settings
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { CORE_MODULES } from './constants';
+
+// Actions
+const SET_OWNER_ID = 'SET_OWNER_ID';
+const SET_SHARING_MANAGEMENT = 'SET_SHARING_MANAGEMENT';
+const SET_SHARED_ROLES = 'SET_SHARED_ROLES';
+const RECEIVE_GET_SHARING_SETTINGS = 'RECEIVE_GET_SHARING_SETTINGS';
+
+const baseInitialState = {
+	sharingSettings: undefined,
+	savedSharingSettings: undefined,
+};
+
+const fetchSaveSharingSettingsStore = createFetchStore( {
+	baseName: 'saveSharingSettings',
+	controlCallback: ( { savedSharingSettings } ) => {
+		return API.set(
+			'core',
+			'modules',
+			'sharing-settings',
+			{ savedSharingSettings },
+			{
+				useCache: false,
+			}
+		);
+	},
+	reducerCallback: ( state, { settings } ) => {
+		return {
+			...state,
+			savedSharingSettings: settings,
+			sharingSettings: settings,
+		};
+	},
+	argsToParams: ( savedSharingSettings ) => ( { savedSharingSettings } ),
+	validateParams: ( { savedSharingSettings } = {} ) => {
+		invariant( savedSharingSettings, 'savedSharingSettings is required.' );
+	},
+} );
+
+const baseActions = {
+	/**
+	 * Sets ownerID for the given modules.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} newOwnerIDs New owner IDs, [moduleSlug]: Number.
+	 * @return {Object} Action for SET_OWNER_ID.
+	 */
+	setOwnerID( newOwnerIDs ) {
+		invariant( newOwnerIDs, 'newOwnerIDs is required.' );
+		return {
+			payload: {
+				newOwnerIDs,
+			},
+			type: SET_OWNER_ID,
+		};
+	},
```

Comment:
By `update each module owner ID using the setOwnerID action` in IB, we meant to call the `setOwnerID` action of individual modules that updates `ownerID` setting. So we don't need this action. What we need to do is to get a store name for each module that receives new owner ID and then dispatch the appropriate `setOwnerID` action using the store name received for the module. There is a dedicated selector in the `CORE_MODULES` datastore to get a store name in the module.